pkg/ddl/notifier/events.go

```diff
@@ -369,14 +385,46 @@ func (s *SchemaChangeEvent) GetRemovePartitioningInfo() (
 	newSingleTableInfo *model.TableInfo,
 	droppedPartInfo *model.PartitionInfo,
 ) {
-	intest.Assert(s.tp == model.ActionRemovePartitioning)
-	return s.oldTableID4Partition, s.tableInfo, s.droppedPartInfo
+	intest.Assert(s.inner.Tp == model.ActionRemovePartitioning)
+	return s.inner.OldTableID4Partition, s.inner.TableInfo, s.inner.DroppedPartInfo
 }
 
 // NewFlashbackClusterEvent creates a schema change event whose type is
 // ActionFlashbackCluster.
 func NewFlashbackClusterEvent() *SchemaChangeEvent {
 	return &SchemaChangeEvent{
-		tp: model.ActionFlashbackCluster,
+		inner: &jsonSchemaChangeEvent{
+			Tp: model.ActionFlashbackCluster,
+		},
+	}
+}
+
+// jsonSchemaChangeEvent is used by SchemaChangeEvent when needed to (un)marshal data,
+// we want to hide the details to subscribers, so SchemaChangeEvent contain this struct.
+type jsonSchemaChangeEvent struct {
+	TableInfo       *model.TableInfo     `json:"tableInfo"`
```

Comment:
1. add `omitempty` JSON tag
2. seems in other places of ddl package we use snake_case

Reply:
PTAL  again https://github.com/pingcap/tidb/pull/56089/commits/be7d271e89af9ef7bb09c8c38156d1b6dc77f007